native/sort.c

```diff
@@ -0,0 +1,74 @@
+#include <stdio.h>
+
+#include "native.h"
+
+typedef struct MapPair{
+    GoString  key;
+    void*     value;
+} MapPair;
+
+// swap elems in MapPair
+void swap(MapPair* lhs, MapPair* rhs) {
+    MapPair temp;
+    temp = *lhs;
+    *lhs = *rhs;
+    *rhs = temp;
+}
+
+int _strcmp(const char *p,const char *q){
```

Comment:
*p and *q may access the invalid memory address without any bound checking.
Maybe add a bound check is better. because we have known the length of key, in GoString.